django/db/models/fields/tuple_lookups.py

```diff
@@ -233,6 +240,23 @@ def check_rhs_elements_length_equals_lhs_length(self):
                 f"must have {len_lhs} elements each"
             )
 
+    def check_rhs_is_query(self):
+        if not (isinstance(self.rhs, Query)):
+            lhs_str = self.get_lhs_str()
+            raise ValueError(
+                f"{self.lookup_name!r} subquery lookup of {lhs_str} "
+                "must be a Query object"
+            )
+
+    def check_rhs_select_length_equals_lhs_length(self):
+        len_lhs = len(self.lhs)
+        if len(self.rhs.select) != len_lhs:
+            lhs_str = self.get_lhs_str()
+            raise ValueError(
+                f"{self.lookup_name!r} subquery lookup of {lhs_str} "
+                f"must have {len_lhs} fields"
+            )
```

Comment:
Similarly here. Perhaps we can report `len(self.rhs.select)`?

Reply:
Okay, I added the `rhs` length to the error message 👍